diff --git a/generator-build/index.js b/generator-build/index.js
index 81ce34d1a863dce3c076bf45a56c560f5bf93cb9..255e697d45e33814a728a6e75f3a4d60fd708167 100644
--- a/generator-build/index.js
+++ b/generator-build/index.js
@@ -7827,7 +7827,8 @@ function buildNFTAnnotation(fileName, relativeOutdir) {
   return `
 // file annotations for bundling tools to include these files
 path.join(__dirname, ${JSON.stringify(pathToPosix(fileName))});
-path.join(process.cwd(), ${JSON.stringify(pathToPosix(relativeFilePath))})`;
+path.join(process.cwd(), ${JSON.stringify(pathToPosix(relativeFilePath))});
+globalThis.SHOULD_BE_UNDEFINED && require(${JSON.stringify("./" + pathToPosix(fileName))});`;
 }
 
 // src/generation/utils/buildRequirePath.ts
