diff --git a/dist/index.mjs b/dist/index.mjs
index fa2d472ffd7be727069ebfdd84f194dcd93aae40..3a97fc2a55c9d6cd269127eac8ee5f95585a00b1 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -75,6 +75,9 @@ async function detectLoader(context, matrix) {
   return null;
 }
 async function isRuntimeSupportsTsx() {
+  if (globalThis.Bun) {
+    return false;
+  }
   if (!nodeVersionNumbers || nodeVersionNumbers[0] < 18 || nodeVersionNumbers[0] === 18 && nodeVersionNumbers[1] < 19 || nodeVersionNumbers[0] === 20 && nodeVersionNumbers[1] < 8) {
     return false;
   }
